lib/Parser/RegexRuntime.cpp

```diff
@@ -2128,9 +2128,28 @@ namespace UnifiedRegex
         int besti = -1;
         CharCount bestMatchOffset = 0;
 
+        if (matcher.literalNextSyncInputOffsets == nullptr)
+        {
+            matcher.literalNextSyncInputOffsets =
+                RecyclerNewArrayLeaf(matcher.recycler, CharCount, ScannersMixin::MaxNumSyncLiterals);
+        }
+        CharCount* literalNextSyncInputOffsets = matcher.literalNextSyncInputOffsets;
```

Comment:
Please add the assert here for numLiterals is lesser than MaxNumSincLiterals.
Rest looks good to me.